src/server/scripts/Commands/cs_tele.cpp

```diff
@@ -343,9 +347,203 @@ class tele_commandscript : public CommandScript
+        if (!handler->extractPlayerTarget(charName, &target, nullptr, &targetName))
+            return false;
+
+        if (!target)
+        {
+            handler->PSendSysMessage(LANG_BAD_VALUE);
+            handler->SetSentErrorMessage(true);
+            return false;
+        }
+
+        // check online security
+        if (handler->HasLowerSecurity(target, ObjectGuid::Empty))
+            return false;
+
+        std::string nameLink = handler->playerLink(targetName);
+        if (target->IsBeingTeleported())
+        {
+            handler->PSendSysMessage(LANG_IS_TELEPORTED, nameLink.c_str());
+            handler->SetSentErrorMessage(true);
+            return false;
+        }
+
+        // -------------
+        // npc target type. 0 = guid, 1 = id, name = 2.
+        int npcTargetType;
+        if (++iter == tokens.end())
+            return false;
+        std::string npcType = *iter;
+        if (npcType == "guid")
+            npcTargetType = 0;
+        else if (npcType == "id")
+            npcTargetType = 1;
+        else if (npcType == "name")
+            npcTargetType = 2;
+        else
+        {
+            handler->SendSysMessage(LANG_BAD_VALUE);
+            handler->SetSentErrorMessage(true);
+            return false;
+        }
+
+        // npc entry (guid, id, or name)
+        if (++iter == tokens.end())
+            return false;
+        std::string rest = *iter;
+        std::string entry = handler->extractKeyFromLink(rest, "Hcreature_entry");
+        if (entry.empty())
+            return false;
+
+        // build query
+        std::ostringstream whereClause;
+        bool isValid = false;
+        switch (npcTargetType)
+        {
+            case 0: // id
+            {
+                uint32 npcId = std::stol(entry);
+                if (isValid = !!npcId)
+                    whereClause << "WHERE id = '" << entry << '\'';
+                break;
+            }
+            case 1: // guid
+            {
+                ObjectGuid::LowType guidLow = std::stoll(entry);
+                if (isValid = !!guidLow)
+                    whereClause << "WHERE guid = '" << guidLow << '\'';
+                break;
+            }
+            case 2: // name
+            {
+                std::string name = entry;
+                if (isValid = (name.size() > 0))
+                {
+                    WorldDatabase.EscapeString(name);
+                    whereClause << ", creature_template WHERE creature.id = creature_template.entry AND creature_template.name LIKE '" << name << '\'';
+                }
+                break;
+            }
+        }
+
+        if (!isValid)
+        {
+            handler->SendSysMessage(LANG_BAD_VALUE);
+            handler->SetSentErrorMessage(true);
+            return false;
+        }
+
+        QueryResult result = WorldDatabase.PQuery("SELECT position_x, position_y, position_z, orientation, map FROM creature %s", whereClause.str().c_str());
```

Comment:
What if the NPC has a waypoint and isn't on that position. Iirc there's a method to retrieve the Creature/Unit entity dynamically without having to query DB.

Reply:
To be honest I pretty much just combined 2 existing gm commands, implementation-wise.
That is, the reason I used this it's because it used in the other gm command - `go creature`.

But if you know another way I'm willing to take a look?

Reply:
Tbf I cannot remember the method that required GUID/Entry MonkaW

Reply:
not ideal but "fine", if its a copy of logic from other commands